artiq/coredevice/comm_kernel.py

```diff
@@ -701,6 +710,15 @@ def _process_async_error(self):
             logger.warning(f"{(', '.join(errors[:-1]) + ' and ') if len(errors) > 1 else ''}{errors[-1]} "
                            f"reported during kernel execution")
 
+    def get_channel_name(self, channel_number):
+        device_db = self.dmgr.get_device_db()
+        for device_name, value in device_db.items():
+            if "arguments" in value:
+                if "channel" in value["arguments"]:
+                    if value["arguments"]["channel"] == channel_number:
+                        return device_name
+        return "unknown"
```

Comment:
This would also potentially incorrectly match on controller entries. Add a check for ``"type": "local"``

Reply:
And maybe even ``module.startswith("artiq.coredevice.")``

Reply:
And perhaps there are even devices that take several RTIO channels and need something more elaborate than this naive code. Please find out.

Reply:
Then that code needs to be shared (cleanly) with the frontend tool and taken out of ``comm_kernel``.

Reply:
> And perhaps there are even devices that take several RTIO channels and need something more elaborate than this naive code. Please find out.

I think the information of the channel address for those device was lost before sending back as the channel number information is right shifted before going to `process_exceptional_status` in `artiq/ksupport/rtio.rs`.
So they are not showing the detail originally.
Do you think it should be identified as well?

Example of the device that take several RTIO channels:
TTLInOut
```
        self.target_o      = (channel << 8) + 0
        self.target_oe     = (channel << 8) + 1
        self.target_sens   = (channel << 8) + 2
        self.target_sample = (channel << 8) + 3
```

Reply:
What would be the use case in front end?

Reply:
Your ``artiq_ch_names`` program (still not convinced this is a good name, but this is how it's called atm).